### PR TITLE
Update Java solution usaco-941

### PR DESCRIPTION
**Pull request by vivianhan404**
so that it matches the one in the video

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
wait is this just changing your previous sol or what
what was wrong w/ the previous one

**Comment by vivianhan404** [comment]
@SansPapyrus683 on the videos thread they asked us to change any solutions to the ones explained in the video editorials so...

**Review by maggie-j-liu — APPROVED** [review]

**Review by nwatx — APPROVED** [review]

File: solutions/usaco-941.mdx [hunk 1]
```diff
@@ -107,72 +107,47 @@ print('yes')
 <JavaSection>
 
 ```java
-import java.util.*;
 import java.io.*;
+import java.util.*;
+
+public class Evolution {
+	static boolean[][] hasChar = new boolean[25][125];
```
Comment (SansPapyrus683) [inline solutions/usaco-941.mdx:114]: put 25 and 125 in a constant

Comment (SansPapyrus683) [reply solutions/usaco-941.mdx:114]: also there's literally no reason for these static variables to be static

File: solutions/usaco-941.mdx [hunk 2]
```diff
@@ -107,72 +107,47 @@ print('yes')
...
-				} else {
-					onlyA = true;
+		br.close();
+
+		boolean works = true;
+		for (int a = 0; a < nxtIdx && works; a++) {
+			for (int b = 0; b < a && works; b++) {
+				int aNum = 0;
+				int bNum = 0;
+				int abNum = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-941.mdx:138]: some comments explaining why you're doing what you're doing would be nice

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
honestly better than your previous code

File: solutions/usaco-941.mdx (hunk 1 above)
Comment (nwatx) [reply solutions/usaco-941.mdx:114]: > also there's literally no reason for these static variables to be static

the code won't compile otherwise?

Comment (SansPapyrus683) [reply solutions/usaco-941.mdx:114]: wait what
why
i don't see them being used in any other methods

Comment (nathangong) [reply solutions/usaco-941.mdx:114]: its being used in the main method, so you need it to be static

File: solutions/usaco-941.mdx [hunk 3]
```diff
@@ -107,72 +107,47 @@ print('yes')
...
-		for (int a = 0; a < characs.length; a++) {
-			for (int b = a + 1; b < characs.length; b++) {
-				if (existsOverlap(characs[a], characs[b], cows)) {
-					out.println("no");
-					out.close();
-					return;
+		BufferedReader br = new BufferedReader(new FileReader("evolution.in"));
+		int popNum = Integer.parseInt(br.readLine());
+		int nxtIdx = 0;
+		for (int i = 0 ; i < popNum; i++) {
```
Comment (nathangong) [inline solutions/usaco-941.mdx:121]:
```suggestion
		for (int i = 0; i < popNum; i++) {
```

File: solutions/usaco-941.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-941.mdx:114]: or you could just you know
put it right in the main method

Comment (nathangong) [reply solutions/usaco-941.mdx:114]: yeah thats an option

**Comment by nathangong** [comment]
Huh there's a weird error that's preventing deployment, @maggie-j-liu do you know what's going on?

**Comment by maggie-j-liu** [comment]
oops i have no idea

**Comment by maggie-j-liu** [comment]
oops sorry i think redeploying might fix the error

**Pull request merged**